package/harvester-os/files/usr/sbin/migration/migrate_harv_mgmt_to_mgmt_br.sh

```diff
@@ -5,40 +5,31 @@ HARV_MGMT="harvester-mgmt"
 IFCFG=
 IFROUTE=
 
+function detect_mgmt () {
+	yq -e '.stages.initramfs[0].files[] | select(.path == "/etc/sysconfig/network/ifcfg-'${HARV_MGMT}'") | .content' "$HARV_CONFIG" > /dev/null
+}
+
 function migrate_mgmt_config () {
 	MODE=
-	# search all files
-	for (( i=0; ; i++ ))
-	do
-		if ! yq -e ".stages.initramfs[0].files[$i]" "$HARV_CONFIG" > /dev/null 2>&1; then
-			echo "end files"
-			break
-		fi
-
-		if yq ".stages.initramfs[0].files[$i].path" "$HARV_CONFIG" | grep -q "/etc/sysconfig/network/ifcfg-${HARV_MGMT}"; then
-			# save config
-			IFCFG=$(yq ".stages.initramfs[0].files[$i].content" "$HARV_CONFIG")
-			echo "$IFCFG"
-		fi
-
-		if yq ".stages.initramfs[0].files[$i].path" "$HARV_CONFIG" | grep -q "/etc/sysconfig/network/ifroute-${HARV_MGMT}"; then
-			# save config
-			IFROUTE=$(yq ".stages.initramfs[0].files[$i].content" "$HARV_CONFIG")
-			echo "$IFROUTE"
-		fi
-	done
+	IFCFG=$(yq '.stages.initramfs[0].files[] | select(.path == "/etc/sysconfig/network/ifcfg-'${HARV_MGMT}'") | .content' "$HARV_CONFIG")
+	IFROUTE=$(yq '.stages.initramfs[0].files[] | select(.path == "/etc/sysconfig/network/ifroute-'${HARV_MGMT}'") | .content' "$HARV_CONFIG")
+	echo "$IFCFG"
```

Comment:
I thought that should use `printf "%s\n" "$IFCFG"`? That would be consistent with later operations.

Reply:
fix